### DISCUSSION TITLE
Is it possible to explicitly disable lazy='subquery'?

### DISCUSSION THREAD

**Discussion by jshen28** [sqlalchemy/sqlalchemy#8024] [Usage Questions]
Hello,  I would like to ask if it is possible to explicitly disable a `lazy='subquery'` relationship loading (for example [here](https://github.com/openstack/neutron/blob/b12a6a07473c777e015cb9f387a93d3e51ff5659/neutron/plugins/ml2/models.py#L131)). 

Here in my opinion DistributedPortBindings is backref to Port and wherever there is a query for port, distributedportbindings are also queried as well which is not necessary for every case. So I would like to ask if it is possible to disable this loading explicitly if it is not needed?

Thank you .

Best Regards,
Norman Shen

**Comment by CaselIT** [comment]
Hi.

Please refer to https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html lazyload seems to be what you are looking for
You can specify a different loading mode in the `options` of a query or select.

**Reply by jshen28** [reply]
thank you for reply. I am trying to find a way to override lazy load mode defined in model. the reason for this, is that I just want to make the modifications simpler. For example, https://github.com/openstack/neutron/blob/b12a6a07473c777e015cb9f387a93d3e51ff5659/neutron/objects/router.py#L117 in this link, there is no explicit `options`, but it indeed still triggers a subquery to load distributedportbindings. My question is is it possible to override  this subquery only by modifying L117 without touching data model..

**Reply by CaselIT** [reply]
> I am trying to find a way to override lazy load mode defined in model

that's what I was replying too. again please refer to that documentation page...
the second section in the page I linked to is this https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html#relationship-loading-with-loader-options that has examples at how to specify the loading technique for relationships

**Reply by jshen28** [reply]
ahh.. I find something called `enable_eagerloads` and I write a demo like this

```python
binding_model = rb_model.RouterL3AgentBinding
sub_query = (admin_ctx.session.query(
    binding_model.router_id,
    func.count(binding_model.router_id).label('count')).enable_eagerloads(False).
             join(l3_attrs.RouterExtraAttributes,
                  binding_model.router_id ==
                  l3_attrs.RouterExtraAttributes.router_id).
             join(l3.Router).
             group_by(binding_model.router_id).subquery())
# sub_query.enable_eagerloads(False)

query = (admin_ctx.session.query(l3.Router, sub_query.c.count).
         outerjoin(sub_query))
```

But looks like it still loads `DistributedPortBindings`, what am I doing wrong? thank you.
